pkg/daemon/drain.go

```diff
@@ -39,12 +39,21 @@ func (dn *Daemon) cordonOrUncordonNode(desired bool) error {
 	}
 	var lastErr error
 	if err := wait.ExponentialBackoff(backoff, func() (bool, error) {
+		// Log has been added to ensure that MCO is correctly performing cordon/uncordon.
+		// This should help us with debugging bugs like https://bugzilla.redhat.com/show_bug.cgi?id=2022387
+		glog.Infof("Initiating %s on node (currently unschedulable: %t)", verb, dn.node.Spec.Unschedulable)
```

Comment:
Sorry, I'm a bit confused about the log line here. Would we see either: `(currently unschedulable: false)` if its scheduleable and `(currently unschedulable: true)` if its cordoned?

Maybe it would be easier to interpret if we used either:
1. use cordon (currently cordoned: true/false) or 
2. flip that, as in something like: `currently scheduleable, !dn.node.Spec.Unschedulable`

Just curious, would we ever cordon the node if it was already cordoned? I guess only if there was a manual update of some sort?

Reply:
yeah, in general workflow MCO won't be try to cordon an already cordoned node. But like you said it can happen when someone either manually cordon it or could be possible when an update failed and we retry the whole update() function. In any case, cordoning an already cordoned node would not lead to any undesired behavior. Also, main purpose of the log is to capture the current state. Perhaps, we can be smart here to skip unnecessary call to CordonUncordon when node is in desired state. Will look into it.

Reply:
About using "currently unschedulable: true" , yeah we can make it more readable if it provides better clarity. I guess, I literally used the name that was present in the node object spec :laughing:

Reply:
updated log message to "currently schedulable:". After some thought, I am not convinced to make CordonUncordon more smarter(i.e. skipping when already in desired state) because that would mean inconstancy in log and event that we send once cordon/uncordon occurs. We would be emitting event without actually trying to cordon/uncordon.